packages/background/src/controllers/transactions/ContractSignatureParser.ts

```diff
@@ -203,18 +203,22 @@ export class ContractSignatureParser {
             return undefined;
         }
 
-        const request = await axios.get(`${etherscanAPI}/api`, {
-            params: {
+        const request = await httpClient.get<{
+            status: string;
+            result: string;
+        }>(
+            `${etherscanAPI}/api`,
+            {
                 module: 'contract',
                 action: 'getabi',
                 address,
                 //TODO: current .env API key is invalid, should we get a new one?
```

Comment:
I think we should implement API keys to these services as we're exceeding the free quote use very quickly. Although, we should also find a better solution to prevent having our keys exposed because we'd probably use more and more external services with this kind of authentication and we would be making our keys public.

To sum up, the TODO can be removed as I've already created cards for this but should be kept in mind.